src/Geno/C++/Components/Project.cpp

```diff
@@ -17,82 +17,107 @@
 
-Project::Project( std::filesystem::path Location )
-	: m_Location( std::move( Location ) )
-	, m_Name    ( "MyProject" )
+File::File( std::filesystem::path Location, std::string Name )
+	: INode( std::move( Location ), std::move( Name ), NodeKind::File )
 {
-	NewFileFilter( "" ); // Create Default Empty FileFilter
+} // File
 
-} // Project
+//////////////////////////////////////////////////////////////////////////
+
+void File::Rename( std::string Name )
+{
+	const std::filesystem::path OldPath = m_Location / m_Name;
+
+	if( std::filesystem::exists( OldPath ) )
+	{
+		const std::filesystem::path NewPath = m_Location / Name;
+		std::filesystem::rename( OldPath, NewPath );
+	}
+
+	m_Name = std::move( Name );
+	m_pParent->SortChildren();
+
+} // Rename
 
 //////////////////////////////////////////////////////////////////////////
 
-Project::Project( Project&& rrOther )
+FileFilter::FileFilter( std::string Name )
+	: INode( {}, std::move( Name ), NodeKind::FileFilter )
 {
-	*this = std::move( rrOther );
+} // FileFilter
 
-} // Project
+//////////////////////////////////////////////////////////////////////////
+
+void FileFilter::Rename( std::string Name )
+{
+	m_Name = std::move( Name );
+	m_pParent->SortChildren();
+
+} // Rename
 
 //////////////////////////////////////////////////////////////////////////
 
-Project& Project::operator=( Project&& rrOther )
+void FileFilter::NewFile( std::filesystem::path Location, std::string Name )
 {
-	m_Kind               = rrOther.m_Kind;
-	m_Location           = std::move( rrOther.m_Location );
-	m_Name               = std::move( rrOther.m_Name );
-	m_FileFilters        = std::move( rrOther.m_FileFilters );
-	m_IncludeDirectories = std::move( rrOther.m_IncludeDirectories );
-	m_LibraryDirectories = std::move( rrOther.m_LibraryDirectories );
-	m_Defines            = std::move( rrOther.m_Defines );
-	m_Libraries          = std::move( rrOther.m_Libraries );
-	m_FilesLeftToBuild   = std::move( rrOther.m_FilesLeftToBuild );
-	m_FilesToLink        = std::move( rrOther.m_FilesToLink );
+	if( !ChildByName( Name ) )
+	{
+		std::ofstream OutputFile( Location / Name, std::ios::binary | std::ios::trunc );
+		if( OutputFile.is_open() )
+		{
+			AddChild( new File( std::move( Location ), std::move( Name ) ) );
+			OutputFile.close();
+		}
+	}
 
-	rrOther.m_Kind       = Kind::Unspecified;
+} // NewFile
 
-	return *this;
+//////////////////////////////////////////////////////////////////////////
 
-} // operator=
+Project::Project( std::filesystem::path Location, std::string Name )
+	: INode( std::move( Location ), std::move( Name ), NodeKind::Project )
+{
+	AddChild( new FileFilter( "" ) ); // Create Default Empty FileFilter
+
+} // Project
 
 //////////////////////////////////////////////////////////////////////////
 
 void Project::Build( ICompiler& rCompiler )
 {
 	m_FilesLeftToBuild.clear();
-	m_FilesToLink     .clear();
+	m_FilesToLink.clear();
```

Comment:
Please don't format code you didn't change! A tip is to look at the changes in the files before staging and committing them. In most git clients nowadays you can stage/unstage/discard individual lines in files!